src/Tribe/Meta/Chunker.php

```diff
@@ -0,0 +1,811 @@
+<?php
+
+/**
+ * Class Tribe__Meta__Chunker
+ *
+ * Chunks large meta to avoid killing the database in queries.
+ *
+ * Databases can have a `max_allowed_packet` value set as low as 1M; we often need to store large blobs of data way
+ * over that and doing so would kill the database ("MySQL server has gone away"...); registering a meta to be chunked
+ * if needed avoids that.
+ *
+ * Example usage:
+ *
+ *      $chunker = tribe( 'chunker' );
+ *      $chunker->register_for_chunking( $post_id, 'my_meta_key' );
+ *
+ *      // ... some code later...
+ *
+ *      // data will be transparently chunked if needed...
+ *      update_meta( $post_id, 'my_meta_key', $some_looooooooooong_string );
+ *
+ *      // ...and glued back together when reading
+ *      get_post_meta( $post_id, 'my_meta_key', true );
+ *
+ * By default the Chunker supports the `post` post type only, filter the `tribe_meta_chunker_post_types` to add yours:
+ *
+ *      add_filter( 'tribe_meta_chunker_post_types`, 'my_chunkable_post_types' );
+ *      function my_chunkable_post_types( $post_types ) {
+ *          $post_types[] = 'book';
+ *
+ *          return $post_types;
+ *      }
+ *
+ * or filter the `tribe_meta_chunker_post_types` filter.
+ */
+class Tribe__Meta__Chunker {
+	public $id;
+	/**
+	 * @var array The cache that will store chunks to avoid middleware operations from fetching the database.
+	 */
+	protected $chunks_cache = array();
+
+	/**
+	 * @var string The separator that's used to mark the start of each chunk.
+	 */
+	protected $chunk_separator = '{{{TCSEP}}}';
+
+	/**
+	 * @var array The post types supported by the Chunker.
+	 */
+	protected $post_types = array( 'post' );
+
+	/**
+	 * @var int The filter priority at which Chunker will operate on meta CRUD operations.
+	 */
+	protected $filter_p = - 1;
```

Comment:
Optional, personal pref thing - but I'd rather see `$filter_priority` than `$filter_p`. No particular need to change it, though, if this was already reviewed and approved in a different PR.